### ISSUE TITLE
peerDependencies issues

### ISSUE THREAD

**Issue by ForestSpark** [Unistyles-OSS/uniwind#134]
Hi, first of all, thank you for the package and your effort! It's awesome to see there is a new library that's making styling React Native apps a breeze.

We're using your library in our app, and we've run into some peerDependency warnings. The issue is that we do not use any of these packages ourselves, so I believe they might need to be corrected in this package itself (specifically - chalk, postcss and lightningcss).

I've run it through Claude and here's what it said about it. Hope it helps.

---

```

### Summary

The current `peerDependencies` in `packages/uniwind/package.json` have several issues that may cause unnecessary installation errors.

---

### Problems Identified

#### 1. postcss not used ❌

**Current:**
```json
"postcss": ">=8"

Problem: Searched the entire codebase - postcss is not imported or used anywhere.

Recommendation: Remove from peerDependencies

---
2. chalk and lightningcss should be optional ⚠️

Current:
"chalk": ">=2",
"lightningcss": ">=1.30.0"

Problem: These are only used in the metro plugin (uniwind/metro):
- chalk - only in src/metro/logger.ts
- lightningcss - only in src/metro/processor/* files

Users who don't use the metro plugin shouldn't need these dependencies.

Recommendation: Mark as optional peerDependencies OR move to regular dependencies

---
Suggested Fix

Option 1: Mark as optional (recommended)

{
  "peerDependencies": {
    "react": ">=19.0.0",
    "react-native": ">=0.81.0",
    "tailwindcss": ">=4",
    "chalk": ">=2",
    "lightningcss": ">=1.30.0"
  },
  "peerDependenciesMeta": {
    "chalk": {
      "optional": true
    },
    "lightningcss": {
      "optional": true
    }
  }
}

Option 2: Move to dependencies

Move chalk to regular dependencies (it's only ~100KB) since it's an internal implementation detail.

---
Impact

- Users without the metro plugin will get peer dependency warnings/errors unnecessarily
- Unnecessary postcss dependency confuses users

---
Files Referenced

- packages/uniwind/package.json (lines 76-83)
- src/metro/logger.ts (chalk usage)
- src/metro/processor/processor.ts (lightningcss usage)

**Comment by Brentlok** [comment]
I agree with `postcss`, we've must left it by mistake when experimenting with alpha versions of uniwind.

You can't really use uniwind without metro plugin so lightningcss and chalk are required. lightningcss is a dependency of tailwind so it should be installed in your project anyway, and I believe that chalk is a dependency of react native, but I may be mistaken there.

Could you share exact warnings that you've got?

**Comment by mikkomcmenamin** [comment]
I think an error I got is related to this. 

`Error: Found config at /Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/apps/native/metro.config.js that could not be loaded with Node.js.
Error: Found config at /Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/apps/native/metro.config.js that could not be loaded with Node.js.
    at loadConfigFile (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/metro-config/src/loadConfig.js:263:15)
    at resolveConfig (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/metro-config/src/loadConfig.js:94:10)
    at loadMetroConfigAsync (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/@expo/cli/src/start/server/metro/instantiateMetro.ts:97:
21)
    at instantiateMetroAsync (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/@expo/cli/src/start/server/metro/instantiateMetro.ts:20
3:7)
    at MetroBundlerDevServer.startImplementationAsync (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/@expo/cli/src/start/server/met
ro/MetroBundlerDevServer.ts:1051:69)
    at MetroBundlerDevServer.startAsync (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/@expo/cli/src/start/server/BundlerDevServer.
ts:160:18)
    at DevServerManager.startAsync (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/@expo/cli/src/start/server/DevServerManager.ts:18
4:7)
    at startAsync (/Users/mikkomcmenamin/Development/prettyokayapps/repos/vibe-training/node_modules/@expo/cli/src/start/startAsync.ts:100:3)
error: script "dev" exited with code 1 `

this fixed it in package.json: 
	"overrides": {
		"chalk": "4.1.2"
	},

Claude Code thinks "The issue is that uniwind/metro is trying to require('chalk'), but you have a newer version of chalk that's ES Module only"

**Comment by ForestSpark** [comment]
@Brentlok This is the peerDependency warning I’m getting. 

```
➤ YN0002: │ expo-app@workspace:. doesn't provide chalk (pddc7f8), requested by uniwind.
➤ YN0002: │ expo-app@workspace:. doesn't provide lightningcss (pb5ea93), requested by uniwind.
➤ YN0002: │ expo-app@workspace:. doesn't provide postcss (pca9e77), requested by uniwind.
```

I see you already fixed the issue with PostCSS and Chalk. Thank you.

When it comes to `lightningcss`, I think it should also be a direct dependency of this package. Looking at Metro and Tailwind, they also have it as a direct dependency. 

https://github.com/expo/expo/blob/451e17c63b74260e2ed0b36667ec03a434f74196/packages/%40expo/metro-config/package.json#L54

https://github.com/tailwindlabs/tailwindcss/blob/dc6a3ce349a082e6c437288e52d0eea9a9ea4ff2/packages/%40tailwindcss-node/package.json#L43
